### ISSUE TITLE
Make U matrices not persistent to reduce `state_dict` size.

### ISSUE THREAD

**Issue by pfebrer** [ACEsuit/mace#124]
**Is your feature request related to a problem? Please describe.**
We are using make with max angular momentum 4 and correlation 3 and the checkpoint files are huge because the U matrices of the `Contraction` class are stored in it (~400MB when the parameters of the model only occupy ~5MB).

**Describe the solution you'd like**
We would like that U matrices weren't stored in the checkpoint file.

**Describe alternatives you've considered**
We think that passing `persistent=False` in this line: https://github.com/ACEsuit/mace/blob/44b6da4a5edaa4c3ef867a11728555403d1d475d/mace/modules/symmetric_contraction.py#L110 should solve the problem and doesn't cause any harm to the model. Otherwise maybe allowing the user to choose whether they are persistent or not would be nice as well.

**Comment by ilyes319** [comment]
Hey @pfebrer,

That is a reasonable solution. Could you try it and check if it solves your issues?

**Comment by pfebrer** [comment]
It indeed solved the problem, my checkpoint files size has been reduced from 680 MB to 26 MB :) I also could restart the training without problems.

But maybe in some case it is useful to store them precomputed?

**Comment by pfebrer** [comment]
I now noticed that this change breaks compatibility with loading models. 

E.g.: if you load a model that stored the U matrices in a version of mace that has them set to `persistent=False` this will result in an error. And the same happens in the opposite case.

**Comment by peterbjorgensen** [comment]
> I now noticed that this change breaks compatibility with loading models.
> 
> E.g.: if you load a model that stored the U matrices in a version of mace that has them set to `persistent=False` this will result in an error. And the same happens in the opposite case.

This could maybe be fixed with non-strict loading. The torch load function has a keyword `strict=False`, but it might be better to just break the backwards compatibility or explicitly remove the U matrices from old checkpoints.

**Comment by pfebrer** [comment]
@ilyes319 do you think you can make them `persistent=False`? To load an old model on the new implementation it would just be a matter of "cleaning" the checkpoint file, i.e. removing the matrices from it.

**Comment by pfebrer** [comment]
Yes, if it can be configured from an argument of the `SymmetricContraction` module (not just `Contraction`) I think it would be fine for us :+1:

**Comment by pfebrer** [comment]
Could we add this? :) (a `persistent_U_matrices` or something similatr argument to `SymmetricContraction` that defaults to `True`) 

I can submit a PR.
